src/MSRIOGroup.cpp

```diff
@@ -226,6 +201,79 @@ namespace geopm
         }
     }
 
+    void MSRIOGroup::register_frequency_signals(void)
+    {
+        if (m_hwp_is_enabled){
+            register_signal_alias("CPU_FREQUENCY_MIN_CONTROL", "MSR::HWP_REQUEST:MINIMUM_PERFORMANCE");
+            register_signal_alias("CPU_FREQUENCY_DESIRED_CONTROL", "MSR::HWP_REQUEST:DESIRED_PERFORMANCE");
+            register_signal_alias("CPU_FREQUENCY_MAX_CONTROL", "MSR::HWP_REQUEST:MAXIMUM_PERFORMANCE");
+            register_signal_alias("PACKAGE_FREQUENCY_MIN_CONTROL", "MSR::HWP_REQUEST_PKG:MINIMUM_PERFORMANCE");
+            register_signal_alias("PACKAGE_FREQUENCY_DESIRED_CONTROL", "MSR::HWP_REQUEST_PKG:DESIRED_PERFORMANCE");
+            register_signal_alias("PACKAGE_FREQUENCY_MAX_CONTROL", "MSR::HWP_REQUEST_PKG:MAXIMUM_PERFORMANCE");
+        } else {
+            register_signal_alias("CPU_FREQUENCY_CONTROL_MAX", "MSR::PERF_CTL:FREQ");
+        }
+        register_signal_alias("FREQUENCY", "MSR::PERF_STATUS:FREQ"); // TODO: Remove @ v2.0
+        register_signal_alias("CPU_FREQUENCY_STATUS", "MSR::PERF_STATUS:FREQ");
+        std::string max_turbo_name;
+        switch (m_cpuid) {
+            case MSRIOGroup::M_CPUID_KNL:
+                max_turbo_name = "MSR::TURBO_RATIO_LIMIT:GROUP_0_MAX_RATIO_LIMIT";
+                break;
+            case MSRIOGroup::M_CPUID_SNB:
+            case MSRIOGroup::M_CPUID_IVT:
+            case MSRIOGroup::M_CPUID_HSX:
+            case MSRIOGroup::M_CPUID_BDX:
+                max_turbo_name = "MSR::TURBO_RATIO_LIMIT:MAX_RATIO_LIMIT_1CORE";
+                break;
+            case MSRIOGroup::M_CPUID_SKX:
+            case MSRIOGroup::M_CPUID_ICX:
+                max_turbo_name = "MSR::TURBO_RATIO_LIMIT:MAX_RATIO_LIMIT_0";
+                break;
+            default:
+                throw Exception("MSRIOGroup: Unsupported CPUID",
+                                GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
+        }
+
+        if (m_hwp_is_enabled)
+        {
+            register_signal_alias("FREQUENCY_MAX", "MSR::HWP_CAPABILITIES:HIGHEST_PERFORMANCE"); // TODO: Remove @ v2.0
+            register_signal_alias("CPU_FREQUENCY_MAX", "MSR::HWP_CAPABILITIES:HIGHEST_PERFORMANCE");
+
+            register_signal_alias("FREQUENCY_MIN", "MSR::HWP_CAPABILITIES:LOWEST_PERFORMANCE"); // TODO: Remove @ v2.0
+            register_signal_alias("CPU_FREQUENCY_MIN", "MSR::HWP_CAPABILITIES:LOWEST_PERFORMANCE");
+
+            register_signal_alias("FREQUENCY_GUARANTEED", "MSR::HWP_CAPABILITIES:GUARANTEED_PERFORMANCE"); // TODO: Remove @ v2.0
+            register_signal_alias("CPU_FREQUENCY_EFFICIENT", "MSR::HWP_CAPABILITIES:MOST_EFFICIENT_PERFORMANCE");
+        }
+        else {
+            register_signal_alias("FREQUENCY_MAX", max_turbo_name); // TODO: Remove @ v2.0
+            register_signal_alias("CPU_FREQUENCY_MAX", max_turbo_name);
```

Comment:
Should be pulled from CPU ID as well, given that the MSR TRL Table is a control.
```suggestion
            register_signal_alias("CPU_FREQUENCY_MAX_CONTROL", max_turbo_name);
```